integration-tests/Main.hs

```diff
@@ -82,12 +81,15 @@ instance NormalizeDefaultExprs Table where
   normalizeDefaultExprs t = t { tableColumns = Map.map normalizeDefaultExprs (tableColumns t)}
 
 instance NormalizeDefaultExprs Column where
-  normalizeDefaultExprs c = c { columnConstraints = Set.map normalizeDefaultExprs (columnConstraints c) }
+  normalizeDefaultExprs c = c { columnConstraints = normalizeDefaultExprs (columnConstraints c) }
 
-instance NormalizeDefaultExprs ColumnConstraint where
+instance NormalizeDefaultExprs ColumnConstraints where
+  normalizeDefaultExprs c =  c { columnDefault = fmap normalizeDefaultExprs (columnDefault c) }
+
+instance NormalizeDefaultExprs DefaultConstraint where
   normalizeDefaultExprs = \case
-    NotNull -> NotNull
-    Default t -> Default (normalizeDefaultExpr t)
+    DefaultExpr t -> DefaultExpr $ normalizeDefaultExpr t
+    Autoincrement mseq -> Autoincrement mseq
```

Comment:
I'm a little confused about why autoincrement is special.  Aren't they also just regular expressions?

Reply:
the specialness is mostly down to the postgres column ownership concept for sequences.  In postgres, sequences can optionally be owned by a column;  if the owning column of a sequence is dropped, the sequence is automatically dropped, too.

I don't think that's too critically important, other than the fact that in beam, SqlSerial is somewhat implied to mean `serial` or `bigserial` column type, which automatically creates an owned sequence of some indeterminate name.  

There are potential reasons why someone might not want this all the time, though; sequences can be used to get values before the corresponding row is created, or the same sequence could be used for several tables. 

These two "ways" of thinking about sequences are encoded in BA types as the AutoIncrement column constraint and the separate Sequence schema map.

a sequence might be known wanted by virtue of the `SqlSerial` but not have a known name (so that it may be added to the Sequences map in Schema), because it hasn't been crated yet;  alternatively, a sequence might be known of, but not have an owner (and may even not appear in any default constraint), so we still want to have the map of sequences.

If SqlSerial was not a thing at all, then I think the special treatment of AutoIncrement is probably also unneeded, but as long as BA needs to support SqlSerial, the special autoincrement feature is neccessary to encode the "unknown/uncreated yet" sequences owned by the autoincrement column.